src/audio/module_adapter/module_adapter.c

```diff
@@ -171,9 +172,13 @@ int module_adapter_prepare(struct comp_dev *dev)
 	 * parameter from sink buffer is settled, and still prior to all references to period_bytes.
 	 */
 	sink = list_first_item(&dev->bsink_list, struct comp_buffer, source_list);
-	mod->period_bytes = audio_stream_period_bytes(&sink->stream, dev->frames);
+	sink_c = buffer_acquire(sink);
+
+	mod->period_bytes = audio_stream_period_bytes(&sink_c->stream, dev->frames);
 	comp_dbg(dev, "module_adapter_prepare(): got period_bytes = %u", mod->period_bytes);
 
+	buffer_release(sink_c);
+
```

Comment:
It looks like here are two different sparse warnings: add `buffer_acquire/buffer_release` and add sparse annotations. I would really like to see these in separate patches.

Reply:
ok, divided into 3 patches now.